### ISSUE TITLE
What is the difference between Socket.io Redis adapter and Redis emitter

### ISSUE THREAD

**Issue by amjedomar** [socketio/socket.io-redis-emitter#111]
While I'm reading this [article][1] from the [Socket.io][2] documentation

I found that the following two packages `@socket.io/redis-adapter` and `@socket.io/redis-emitter` are used to emit data to the clients that are on the other servers.

Are these two packages different? And if yes what is the difference? And when to use one instead of the other?

  [1]: https://socket.io/docs/v4/redis-adapter
  [2]: https://socket.io

**Comment by amjedomar** [comment]
The answer to this question exists on StackOverflow: https://stackoverflow.com/questions/71578212/what-is-the-difference-between-socket-io-redis-adapter-and-redis-emitter/71602798#71602798

**Comment by amjedomar** [comment]
Thanks, @darrachequesne for the answer.

I have spent hours to figuring the difference between the two. I know the difference seems quite easy to notice but I was confused the first time.

I had already answered my question that I asked at StackOverflow: https://stackoverflow.com/questions/71578212/what-is-the-difference-between-socket-io-redis-adapter-and-redis-emitter/71602798#71602798

But I liked your answer as it is very short.
> The adapter is a component inside the Socket.IO server, while the emitter can be used in another process/service.

So I quoted it and added it to my answer at StackOverflow to help future people know the difference.
